src/models/capture_group_patterns.rs

```diff
@@ -38,12 +39,20 @@ impl CGPattern {
   pub(crate) fn pattern(&self) -> String {
     self.0.to_string()
   }
+
+  pub(crate) fn extract_regex(&self) -> String {
+    let mut _val = &self.pattern()[4..];
+    _val.to_string()
+  }
 }
 
 impl Validator for CGPattern {
   fn validate(&self) -> Result<(), String> {
     if self.pattern().starts_with("rgx ") {
-      panic!("Regex not supported")
+      let mut _val = &self.pattern()[4..];
```

Comment:
Why do we have this logic here and in `extract_regex()` both? This probably should not be duplicated in case we ever change the identifier, and also should be something like `len(REGEX_QUERY_PREFIX)` and that prefix used elsewhere to match `"rgx "`